aws-sns-topicpolicy/src/main/java/software/amazon/sns/topicpolicy/DeleteHandler.java

```diff
@@ -27,29 +23,44 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final Logger logger) {
 
         this.logger = logger;
+        ResourceModel resourceModel = request.getDesiredResourceState();
 
-        final ResourceModel model = request.getDesiredResourceState();
-        // primary id must be set up
-        if (StringUtils.isNullOrEmpty(model.getId())
-                || CollectionUtils.isNullOrEmpty(model.getTopics()))
-        {
-            throw new CfnInvalidRequestException("Invalid delete request, must contain Primary Id.");
+        if(CollectionUtils.isNullOrEmpty(resourceModel.getTopics())){
+            return ProgressEvent.failed(resourceModel, callbackContext, null, "Value of property Topics must be of type List of String");
         }
 
-        return ProgressEvent.progress(model, callbackContext)
-                .then(progress -> doDelete(proxy, proxyClient, request, progress))
-                .then(progress -> ProgressEvent.<ResourceModel, CallbackContext> builder()
-                        .status(OperationStatus.SUCCESS)
-                        .build());
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Delete TopicPolicy", request.getStackId(), request.getClientRequestToken()));
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress -> Delete(proxy, proxyClient, request, progress, logger))
+                .then(progress -> ProgressEvent.defaultSuccessHandler(null));
     }
 
-    private ProgressEvent<ResourceModel, CallbackContext> doDelete(
+    protected ProgressEvent<ResourceModel, CallbackContext> Delete(
             final AmazonWebServicesClientProxy proxy,
             final ProxyClient<SnsClient> proxyClient,
             final ResourceHandlerRequest<ResourceModel> request,
-            final ProgressEvent<ResourceModel, CallbackContext> progress) {
-        List<String> topics = request.getDesiredResourceState().getTopics();
-        return handleDelete(proxy, proxyClient, request, progress, topics, ACTION_DELETED, DELETE_HANDLER , logger);
+            ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+        final CallbackContext callbackContext = progress.getCallbackContext();
+        List<String> topics = model.getTopics();
+        for (final String topicArn : topics) {
```

Comment:
how are we accounting for this propertyresourcelock: https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/topicPolicy/TopicPolicyDeleteWorkflowImpl.java#:~:text=public%20Promise%3C,(wait)%3B

Reply:
Skip Lock for now.

Reply:
Double checked with Andrew. Property resource locking is already handled for all of the affected resources, but we should make sure during integration testing.